packages/wallets/exodus/src/adapter.ts

```diff
@@ -0,0 +1,268 @@
+import {
+    BaseMessageSignerWalletAdapter,
+    EventEmitter,
+    SendTransactionOptions,
+    WalletAccountError,
+    WalletConnectionError,
+    WalletDisconnectedError,
+    WalletDisconnectionError,
+    WalletError,
+    WalletName,
+    WalletNotConnectedError,
+    WalletNotReadyError,
+    WalletPublicKeyError,
+    WalletReadyState,
+    WalletSignTransactionError,
+    WalletWindowClosedError,
+    scopePollingDetectionStrategy,
+} from '@solana/wallet-adapter-base';
+import { Connection, PublicKey, SendOptions, Transaction, TransactionSignature } from '@solana/web3.js';
+
+interface ExodusWalletEvents {
+    connect(...args: unknown[]): unknown;
+    disconnect(...args: unknown[]): unknown;
+}
+
+interface ExodusWallet extends EventEmitter<ExodusWalletEvents> {
+    isExodus?: boolean;
+    publicKey?: { toBytes(): Uint8Array };
+    isConnected: boolean;
+    signTransaction(transaction: Transaction): Promise<Transaction>;
+    signAllTransactions(transactions: Transaction[]): Promise<Transaction[]>;
+    signAndSendTransaction(
+        transaction: Transaction,
+        options?: SendOptions
+    ): Promise<{ signature: TransactionSignature }>;
+    signMessage(message: Uint8Array): Promise<{ signature: Uint8Array }>;
+    connect(): Promise<void>;
+    disconnect(): Promise<void>;
+    _handleDisconnect(...args: unknown[]): unknown;
+}
+
+interface ExodusWindow extends Window {
+    solana?: ExodusWallet;
```

Comment:
Can you attach the object to `window.exodus`? We aren't accepting new adapters that make use of `window.solana`, which hurts our ability to reliably detect all wallets. Wallets that already use it have been asked to migrate. Medium term, the plan is to push all wallets to implement https://github.com/solana-labs/wallet-adapter/issues/215 and resolve detection issues across the board.

Reply:
> Can you attach the object to window.exodus?

Yes, addressed in f3f4b33. We were doing both `window.solana` and `window.exodus.solana`, but will pass this on to the team doing the injection.